src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_StringInterpolation.cs

```diff
@@ -33,6 +32,42 @@ private BoundExpression RewriteInterpolatedStringConversion(BoundConversion conv
             return result;
         }
 
+        private static bool CanLowerToStringConcatenation(BoundInterpolatedString node)
+        {
+            foreach (var part in node.Parts)
+            {
+                if (part is BoundStringInsert fillin)
+                {
+                    // this is one of the expression holes
+                    if (fillin.HasErrors ||
+                        fillin.Value.Type?.SpecialType != SpecialType.System_String ||
+                        fillin.Alignment != null ||
+                        fillin.Format != null)
+                    {
+                        return false;
+                    }
+                }
+            }
+
+            return true;
+        }
+
+        private static string Unescape(string s)
+        {
+            var builder = PooledStringBuilder.GetInstance();
+            int formatLength = s.Length;
+            for (int i = 0; i < formatLength; i++)
+            {
+                char c = s[i];
+                builder.Builder.Append(c);
```

Comment:
>builder.Builder [](start = 16, length = 15)

Consider extracting local for `builder.Builder`.

Reply:
Should that be done in the existing `MakeInterpolatedStringFormat` too?

Reply:
Yes, that'd be good. Thanks

---
In reply to: [190506216](https://github.com/dotnet/roslyn/pull/26612#discussion_r190506216) [](ancestors = 190506216)